e2e/windows/e2e-scenario.sh

```diff
@@ -125,8 +129,21 @@ tee $SCENARIO_NAME-vmss.json > /dev/null <<EOF
 }
 EOF
 
-jq --arg clientCrt "$clientCertificate" --arg vmssName $DEPLOYMENT_VMSS_NAME 'del(.KubeletConfig."--pod-manifest-path") | del(.KubeletConfig."--pod-max-pids") | del(.KubeletConfig."--protect-kernel-defaults") | del(.KubeletConfig."--tls-cert-file") | del(.KubeletConfig."--tls-private-key-file") | .ContainerService.properties.certificateProfile += {"clientCertificate": $clientCrt} | .PrimaryScaleSetName=$vmssName' nodebootstrapping_config.json > $WINDOWS_E2E_IMAGE-nodebootstrapping_config_for_windows.json
-jq -s '.[0] * .[1]' $WINDOWS_E2E_IMAGE-nodebootstrapping_config_for_windows.json scenarios/$SCENARIO_NAME/$WINDOWS_E2E_IMAGE-property-$SCENARIO_NAME.json > scenarios/$SCENARIO_NAME/$WINDOWS_E2E_IMAGE-nbc-$SCENARIO_NAME.json
+# Save $WINDOWS_E2E_IMAGE-nodebootstrapping_config_for_windows.json from the node bootstrapping config generated from e2e-starter.sh
+# Removed the "network-plugin" tag o.w. kubelet error for 1.24.0+ contains "failed to parse kubelet flag: unknown flag: --network-plugin"
+# "network-plugin" works for 1.23.15 and below (won't see this parsing error in kubelet.err.log)
+jq --arg clientCrt "$clientCertificate" --arg vmssName $DEPLOYMENT_VMSS_NAME \
+'del(.KubeletConfig."--pod-manifest-path") | del(.KubeletConfig."--pod-max-pids") | 
+del(.KubeletConfig."--protect-kernel-defaults") | del(.KubeletConfig."--tls-cert-file") | 
+del(.KubeletConfig."--tls-private-key-file") | del(.KubeletConfig."--network-plugin") | 
+.ContainerService.properties.certificateProfile += {"clientCertificate": $clientCrt} | 
+.PrimaryScaleSetName=$vmssName' nodebootstrapping_config.json > $WINDOWS_E2E_IMAGE-nodebootstrapping_config_for_windows.json
+
+# Produce a finalized managed cluster config (nbc?) given 
+# 1. the node bootstrapping config from e2e-starter (kubeconfig)
+# 2. the vmss cse deployment config generated from this file (windows package \in k8s components)
+jq -s '.[0] * .[1]' $WINDOWS_E2E_IMAGE-nodebootstrapping_config_for_windows.json \
```

Comment:
what is the change in this part? If it is only for format, please change it later in another PR.

Reply:
Removed the "--network-plugin" from "KubeletConfig" in the json. 

A little bit of context:
Found that if the version in the `K8sComponents/WindowsPackageURL` is below 1.24.0 (e.g. 1.23.12, 1.23.15) in `property-windows-template.json`, the deployment node can join the test cluster. Otherwise the pipeline fails and the node cannot join the cluster. (e.g. 1.24.9, 1.24.10)

@junjiezhang1997 later helped me with this last week. I logged onto the deployment VMSS node and checked the `kubelet.err.log` which says `Error: failed to parse kubelet flag: unknown flag: --network-plugin
Usage:
  kubelet [flags]` This was apparently different from @junjiezhang1997 found on a normal `1.24.10` deployment node. There was no such tag in `CustomDataSetupScript.ps1` or `kubelet.err.log` he found.

So we decided to remove it from the json files and confirmed it worked for versions both below and beyond 1.24.0. (*by checking the k8s version of the deployment node $e.g. winXXXXX$ that joined the test cluster on the portal*)